design-proposal.md

```diff
@@ -0,0 +1,66 @@
+# Proposal: Design of the rkt + Kubernetes CRI
+
+## Abstract
+
+This proposal can be viewed as more a discussion than an actual proposal, because we still haven't reached a consensus on the architecture.
+
+## Backgroud
+
+As container runtime interface (CRI) proposals and original stub codes are already merged into Kubernetes, the actual work has already begun.
+All existing and future container runtimes are expected to conform the interface and provides feature parities.
+
+## Design(s)
+
+There are three different designs:
+
+- `Kubelet` --_gPRC calls_--> `rktlet service` --_invoke 'rkt' binary, or use some rkt library_--> result
```

Comment:
I'm also fine on this proposal (looks like a cleaner design), and I'm mostly concerned on two points at the moment:
 * may require some long-due refactoring to effectively use rkt as a library
 * upgrading scenarios may be troublesome when mixing binary/library processes (but it may be avoided if the kubelet sticks to only use one of those).

On the positive side, this would help improving rkt code modularity and avoid sticking with an half-baked api-service.